src/components/AddressSearch.js

```diff
@@ -23,134 +23,125 @@ const propTypes = {
     value: PropTypes.string,
 
     /** A callback function when the value of this field has changed */
-    onChangeText: PropTypes.func.isRequired,
+    onChange: PropTypes.func.isRequired,
 
     /** Customize the ExpensiTextInput container */
     containerStyles: PropTypes.arrayOf(PropTypes.object),
 
     ...withLocalizePropTypes,
 };
+
 const defaultProps = {
     value: '',
     containerStyles: null,
 };
 
-// Do not convert to class component! It's been tried before and presents more challenges than it's worth.
-// Relevant thread: https://expensify.slack.com/archives/C03TQ48KC/p1634088400387400
-// Reference: https://github.com/FaridSafi/react-native-google-places-autocomplete/issues/609#issuecomment-886133839
-const AddressSearch = (props) => {
-    const googlePlacesRef = useRef();
-    const [displayListViewBorder, setDisplayListViewBorder] = useState(false);
-    useEffect(() => {
-        if (!googlePlacesRef.current) {
-            return;
-        }
+class AddressSearch extends PureComponent {
+    constructor(props) {
+        super(props);
 
-        googlePlacesRef.current.setAddressText(props.value);
-    }, []);
+        this.state = {
+            skippedFirstOnChangeText: false,
```

Comment:
Having the ref was more intuitive than this and would recommend we go back to that if possible.

Reply:
In my opinion, I don't think it makes it more intuitive, but I'm ok with changing it. If I change it to ref, this would mean that this component should be a functional component again, right?